### PR TITLE
Update integration test to cover signature checking on device

### PR DESCRIPTION
**Pull request by AlCutter**
- fixes the integration EXPECT_FAIL test which would have caught this on the signature PR.
- adds a "hacker tool" to easily modify proof bundles
- updates integration test with a new test for checking signatures

### CODE REVIEW

File: binary_transparency/firmware/cmd/hacker/modify_bundle/modify_bundle.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	var fs api.FirmwareStatement
+	if err := json.Unmarshal(pb.ManifestStatement, &fs); err != nil {
+		glog.Exitf("Failed to parse FirmwareStatement: %q", err)
+	}
+	var fm api.FirmwareMetadata
+	if err := json.Unmarshal(fs.Metadata, &fm); err != nil {
+		glog.Exitf("Failed to parse FirmwareMetadata: %q", err)
+	}
+
+	if len(*binaryPath) > 0 {
```
Comment (mhutchinson) [inline binary_transparency/firmware/cmd/hacker/modify_bundle/modify_bundle.go:71]: I wonder if it's worth erroring, or at least printing a warning if this isn't true. Without executing this block all we're doing is round tripping the input and serializing it again? Being noisier in this case makes it harder to use this tool wrong.

File: binary_transparency/firmware/integration/ft_test.sh [hunk 2]
```diff
@@ -92,27 +92,43 @@ go run ../cmd/emulator/dummy/ \
...
 ####################
-banner "Fiddle with installed firmware & manifest and try to boot device"
+banner "Replace installed firmware, update manifest hash, and try to boot device"
 HACKED_FIRMWARE=../testdata/firmware/dummy_device/hacked.wasm
 cp -v ${HACKED_FIRMWARE} ${DEVICE_STATE}/firmware.bin
 malware_hash=$(sha512sum ${DEVICE_STATE}/firmware.bin | awk '{print $1}' | xxd -r -p | base64 -w 0 )
 echo "new hash ${malware_hash}"
 mv ${DEVICE_STATE}/bundle.json ${DEVICE_STATE}/bundle.json.orig
 # This beastly jq command unpacks the nested json structures and replaces just the fimware measurement with the one from our hacked firmware:
 jq --arg hacked ${malware_hash} -c '.ManifestStatement=(.ManifestStatement|@base64d|fromjson|.Metadata=(.Metadata|@base64d|fromjson|.ExpectedFirmwareMeasurement=$hacked|tojson|@base64)|tojson|@base64)' ${DEVICE_STATE}/bundle.json.orig > ${DEVICE_STATE}/bundle.json
```
Comment (mhutchinson) [inline binary_transparency/firmware/integration/ft_test.sh:110]: We can replace this with the new command and then drop the requirement on `jq` now?

Comment (AlCutter) [reply binary_transparency/firmware/integration/ft_test.sh:110]: I thought about that and stopped short for "narrative" reasons ("I'll try this... bah", "ok, I'll up my game/steal keys/build tooling", etc. etc.), but on reflection you're right - the shell and jq stuff is pretty opaque, and I can pop the narrative into the comments.

File: binary_transparency/firmware/cmd/hacker/modify_bundle/modify_bundle.go (hunk 1 above)
Comment (AlCutter) [reply binary_transparency/firmware/cmd/hacker/modify_bundle/modify_bundle.go:71]: That's a reasonable thought, and if this tool grows in functionality then we can wind that back if necessary.

File: binary_transparency/firmware/cmd/hacker/modify_bundle/modify_bundle.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+// proof bundle file, optionally sign the firmware statement, and finally write the bundle
+// to the specified output file.
+//
+// Usage:
+//   go run ./cmd/hacker/modify_bundle/ --logtostderr --device=[dummy,usbarmory] --binary=/path/to/new/firmware/image --input=/path/to/bundle.json --output=/path/to/bundle.json
+//
+// The first time you use this tool there will be no prior firmware metadata
+// stored on the device and the tool will fail.  In this case, use the --force
+// flag to apply the update anyway thereby creating the metadata.
+// Subsequent invocations should then work without needing the --force flag.package main
```
[the remark this answers is not in the document]
Comment (AlCutter) [reply binary_transparency/firmware/cmd/hacker/modify_bundle/modify_bundle.go:27]: *cough* guilty as charged, m'lud - removed.

File: binary_transparency/firmware/cmd/hacker/modify_bundle/modify_bundle.go [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+// limitations under the License.
+
+// modify_bundle is a hacker tool for modifying proof bundles.
+//
+// It can update the expected measurement and firmware image hashes of the specified
+// proof bundle file, optionally sign the firmware statement, and finally write the bundle
+// to the specified output file.
+//
+// Usage:
+//   go run ./cmd/hacker/modify_bundle/ --logtostderr --device=[dummy,usbarmory] --binary=/path/to/new/firmware/image --input=/path/to/bundle.json --output=/path/to/bundle.json
```
[the remark this answers is not in the document]
Comment (AlCutter) [reply binary_transparency/firmware/cmd/hacker/modify_bundle/modify_bundle.go:22]: Good call, done

**Review by mhutchinson — APPROVED** [review]

**Pull request merged**